src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -1772,10 +1809,31 @@ public Boolean visitWildcardType(WildcardType t, Type s) {
                     // ---------------------------------------------------------------------------
                     return isSameWildcard(t, s)
                         || isCaptureOf(s, t)
-                        || ((t.isExtendsBound() || isBoundedBy(wildLowerBound(t), wildLowerBound(s),
-                                (t1, s1, w) -> isSubtype(t1, s1, false, SubtypingRelationKind.REF_VAL_ALLOWED))) &&
-                            (t.isSuperBound() || isBoundedBy(wildUpperBound(s), wildUpperBound(t),
-                                (t1, s1, w) -> isSubtype(t1, s1, false, SubtypingRelationKind.REF_VAL_ALLOWED))));
+                        || checkIfBoundedBy(t, s);
+                }
+            }
+
+            boolean checkIfBoundedBy(WildcardType t, Type s) {
+                if (t.isUnbound()) return true;
+                if (t.isExtendsBound()) {
+                    boolean isBoundedBy = isBoundedBy(wildUpperBound(s), wildUpperBound(t),
+                            (t1, s1, w) -> isSubtype(t1, s1, false, SubtypingRelationKind.REF_VAL_ALLOWED));
+                    if (isBoundedBy) {
+                        return true;
+                    }
+                    return (allowUniversalTVars &&
```

Comment:
And, this is made messier by the fact that:

```
val = ref
```

In my example does not trigger a subtying test. It triggers an assignment conversion, which in turn can use the new form of parameterized conversions.

Reply:
> This extra check feels outside the spirit of the rules discussed in https://openjdk.java.net/jeps/8261529. That is, I'd expect type containment to recurse using `isBoundedBy` instead of `isSubtype` (as the first branch of this code does).

just to provide some context, the reason that supported in my view using this extra check comes from the universal tvars jep in particular where it says:

```
Changing a wildcard bound in a parameterized type from a universal type-variable type (T) or a primitive class type (Point) to its reference type (T.ref, Point.ref), or vice versa (where the conversion is not already allowed by subtyping):
```
this is why on a failure of isBoundedBy which at the end is based on subtyping I tried this more general relation. Although I realize from the additional points you mentioned here that these opens questions in other fronts like type well-formedness.